src/System.Device.Gpio/System/Device/Pwm/Channels/PwmChannel.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Device.Pwm
+{
+    /// <summary>
+    /// Represents a single PWM channel.
+    /// </summary>
+    public abstract partial class PwmChannel : IDisposable
```

Comment:
NIT: This type should be under System\Device\Pwm, since we usually echo the namespace structure with the folder structure for the implementation.

Reply:
@joperezr Shouldn't this follow similar to I2C/SPI Devices' folders?  Instead of Devices, it's Channels. I left the namespaces the same just as we did for Devices (System.Device.I2c/System.Device.Spi), as well.  I thought of placing under the Pwm folder, but Channels folder is the grouping of the different channels (and future home of SoftPwmChannel).  Plus, if we ever had to expand in future (the infamous controller, chip, or such 😄), we would be ahead of the game on separating the channels.  If this is the direction, we might need to fix I2C/SPI to be same?

Reply:
but I2cDevice and SpiDevice are still in the I2c (and Spi) root namespace, and they are the types that will be most likely used most of the time, since people will do I2cDevice.Create. Same goes here, I'm fine with having the platform specific channels inside the channels namespace, but PwmChannel should be in the root of Pwm as that one will be the one used by consumers.

Reply:
I'm still scratching my head on this.  See attachment to compare PWM with SPI.

PwmChannel, along with all flavors of Channels, are under the Channels folder.
All flavors of Channels are under the namespace System.Device.Pwm including PwmChannel.

SpiDevice, along with all flavors of Devices, are under the Devices folder.
All flavors of Devices are under the namespace System.Device.Spi including SpiDevice.

You can call SpiDevice.Create(...).
You can call PwmChannel.Create(...).

Am I missing something?  I could be staring at the screen too long 😄 

![PwmVsSpi](https://user-images.githubusercontent.com/2956825/60840896-79178500-a19e-11e9-9ddc-b67093f7586d.png)

Reply:
I see the issue then 😄. For folder structure, we always try to mimic the namespace, so in this case, SpiDevice.cs, and PwmChannel.cs, would need to go one level up in order to be placed where they are supposed to be. I thought PwmChannel was under a Channels namespace, but I just saw that it is directly under Pwm. I'm fine if we want to keep the UnixPwmChannel and the Windows10PwmChannel under the Channels folder (as long as they are also on the Channels namespace) but the idea here is that unless this is the test project, we want the folder structure to match the namespace, does that make sense?

Reply:
We did discuss that for I2c/Spi.  I believe it was [decided to move I2cDevice/SpiDevice to the Devices folder](https://github.com/dotnet/iot/pull/533#discussion_r298669724).

Yes, all channels are in the System.Device.Pwm namespace, but located under the Channels folder.

Reply:
> I'm fine if we want to keep the UnixPwmChannel and the Windows10PwmChannel under the Channels folder (as long as they are also on the Channels namespace)

Okay.  I'll update PWM related in this PR, and will create another PR to move the I2C/SPI related later.

Reply:
Just an opinion... I don't really like this file move.  PwmChannel.Linux.cs and PwmChannel.Windows.cs are partials of PwmChannel.  This means all 3 files have to go up one level.  We get back to mixture of Channel files located in different folders.  This is the same for I2cDevice.cs/I2cDevice.Linux.cs/I2cDevice.Windows.cs and Spi files, as well.

I guess having them separate isn't a biggie, just seems like all flavors would be located together.